ros_ws/src/car_control/src/dms_controller.cpp

```diff
@@ -2,37 +2,65 @@
 
 DMSController::DMSController()
 {
-    this->m_heartbeat_subscriber =
-        this->m_node_handle.subscribe<std_msgs::Int64>(TOPIC_DMS_HEARTBEAT, 1, &DMSController::heartbeatCallback, this);
-    this->m_unlock_motor_publisher = this->m_node_handle.advertise<std_msgs::Bool>(TOPIC_UNLOCK_MOTOR, 1);
+    this->m_heartbeat_manual_subscriber =
+        this->m_node_handle.subscribe<std_msgs::Int64>(TOPIC_HEARTBEAT_MANUAL, 1,
+                                                       &DMSController::heartbeatManualCallback, this);
+    this->m_heartbeat_autonomous_subscriber =
+        this->m_node_handle.subscribe<std_msgs::Int64>(TOPIC_HEARTBEAT_AUTONOMOUS, 1,
+                                                       &DMSController::heartbeatAutonomousCallback, this);
+    this->m_drive_mode_publisher = this->m_node_handle.advertise<std_msgs::Int32>(TOPIC_DRIVE_MODE, 1);
     this->configureParameters();
-    this->m_last_heartbeat_received = std::chrono::steady_clock::time_point::min();
+    this->m_last_heartbeat_manual = std::chrono::steady_clock::time_point::min();
+    this->m_last_heartbeat_autonomous = std::chrono::steady_clock::time_point::min();
 }
 
 void DMSController::spin()
 {
     ros::Rate loop(this->m_update_frequency);
     while (ros::ok())
     {
-        this->publishUnlockMotor();
+        this->publishDriveMode();
         ros::spinOnce();
         loop.sleep();
     }
 }
 
-void DMSController::publishUnlockMotor()
+DriveMode DMSController::getDriveMode()
```

Comment:
IMO the name is ok because the function has no side effects (It doesn't change the state of anything). You can run it assuming that it does nothing other then getting a member. But if you have a good idea for an alternative name, I'll take it.

Reply:
I think its okay here